File: sdk/SDK.Examples/src/GetPackageListExample.cs
using System;
using System.IO;
using Silanis.ESL.SDK;
using Silanis.ESL.SDK.Builder;
using System.Collections.Generic;

namespace SDK.Examples
{
	public class GetPackageListExample : SDKSample
	{
        public static void Main (string[] args)
        {
            new GetPackageExample().Run();
        }

        public GetPackageListExample( Props props ) : this(props.Get("api.key"), props.Get("api.url")) {
        }

        public GetPackageListExample( String apiKey, String apiUrl ) : base( apiKey, apiUrl ) {
        }

        override public void Execute()
        {
			//Get the packages that have status COMPLETED, starting from the most recent package and getting 20 packages per page
			Page<DocumentPackage> packages = eslClient.PackageService.GetPackages (DocumentPackageStatus.COMPLETED, new PageRequest(1, 20));

			while (packages.HasNextPage())
			{
				packages = eslClient.PackageService.GetPackages (DocumentPackageStatus.COMPLETED, packages.NextRequest);
			}
		}
	}
}

File: sdk/SDK.VBExamples/SDK.VBExamples/GetPackageListExample.vb
Imports System
Imports System.IO
Imports Silanis.ESL.SDK
Imports Silanis.ESL.SDK.Builder
Imports System.Collections.Generic

Namespace SDK.VBExamples
    Public Module GetPackageListExample
        Dim apiToken = "YOUR TOKEN HERE"
        Dim baseUrl = "ENVIRONMENT URL HERE"

        Sub Main()
            ' Create new esl client with api token and base url
            Dim client = New EslClient(apiToken, baseUrl)

            'Get the packages that have status COMPLETED, starting from the most recent package and getting 20 packages per page
            Dim packages = client.PackageService.GetPackages(DocumentPackageStatus.COMPLETED, New PageRequest(1, 20))

            PrintPage(packages)

            While (packages.HasNextPage())
                packages = client.PackageService.GetPackages(DocumentPackageStatus.COMPLETED, packages.NextRequest)
                PrintPage(packages)
            End While
        End Sub

        Sub PrintPage(ByRef page)
            Console.WriteLine("Got {0} packages, total = {1}", page.NumberOfElements, page.TotalElements)

            For Each package In page
                Console.WriteLine("Package {0} has status {1}", package.Name, package.Status)
            Next
        End Sub
    End Module
End Namespace